### ISSUE TITLE
Resolve duplicate line warnings in nonnusdionysiaca.xml

### ISSUE THREAD

**Issue by whoopsedesy** [sasansom/sedes#79]
Cf. #77 

As of a58e572eecea5faf7926c71c08974c4f5836750f, there are these warnings:

<pre>
<s>warning: Dion.: duplicate line '25.528'</s>
<s>warning: Dion.: duplicate line '28.83'</s>
<s>warning: Dion.: duplicate line '28.84'</s>
<s>warning: Dion.: duplicate line '28.93'</s>
<s>warning: Dion.: duplicate line '28.94'</s>
<s>warning: Dion.: duplicate line '32.90'</s>
warning: Dion.: duplicate line '37.568'
warning: Dion.: duplicate line '37.625'
<s>warning: Dion.: duplicate line '39.385'</s>
<s>warning: Dion.: duplicate line '40.566'</s>
warning: Dion.: duplicate line '44.145'
</pre>

**Comment by whoopsedesy** [comment]
25.528 was caused by an extraneous `<lb />` empty line, fixed in 286924389ee9ffca3ecc021e20edb7012fe33571.

**Comment by whoopsedesy** [comment]
28.83, 28.84, 28.93, 28.94 look like simple errors. Fixed in https://github.com/sasansom/sedes/commit/efeb4ad5c3b11e9290ee890b2126f852d1f749a3.

https://archive.org/details/dionysiaca02nonnuoft/page/352/mode/1up
![dionysiaca02nonnuoft_0368](https://user-images.githubusercontent.com/53617134/210195161-7d245e5e-4bc2-43c1-b0fe-864a33af9b6c.jpg)

These line numbers are also duplicates in Perseus 5.0:

https://github.com/PerseusDL/canonical-greekLit/blob/812f91f083f88cf789b37be89b22606ca6f27f6c/data/tlg2045/tlg001/tlg2045.tlg001.perseus-grc1.xml
```
<l n="82">Δεξιόχου δὲ φονῆα καλέσσατο θυιάδι φωνῇ,</l>
<l n="83">λοίγιον ὑβριστῆρι χέων ἔπος ἀνθερεῶνι·</l>
<l n="84">‘στῆθι, κύων, μὴ φεῦγε, Κορύμβασε, καί σε διδάξω,</l>
<l n="83">οἷοι ἀκοντιστῆρες ὀπάονές εἰσι Λυαίου. </l>
<l n="84">ὑμέας εἰς Φρυγίην ληίσσομαι, ἄστεα δʼ Ἰνδῶν</l>
<l n="85">δῃώσει δόρυ τοῦτο, καὶ Ἰνδοφόνον μετὰ νίκην</l>
<l n="86">Δηριάδην θεράποντα Διωνύσοιο τελέσσω·</l>
<l n="87">παρθενικὴ δʼ ἀνάεδνος ἑὴν λύσειε κορείην,</l>
<l n="90">δεχνυμένη Σατύροιο δασυστέρνους ὑμεναίους, </l>
<l n="91">Ἰνδὴ Μυγδονίοιο μιαινομένη σχεδόν Ἕρμου.’</l>
<l n="92">ὣς φαμένου κεχόλωτο Κορύμβασος, ὀψιμόθου δὲ</l>
<l n="93">φθεγγομένου Κλυτίοιο διέθρισεν ἀνθερεῶνα·</l>
<l n="94">καὶ κεφαλὴ πεπότητο μετάρσιος ἅλματι Μοίρης,</l>
<l n="93">αἱμαλέῃ ῥαθάμιγγι περιρραίνουσα κονίην. </l>
<l n="94">καὶ νέκυν ὀρχηστῆρα παλινδίνητον ἐάσας</l>
<l n="95">Σειληνοὺς ἐφόβησε Κορύμβασος, ἔξοχος Ἰνδῶν,</l>
```

**Comment by whoopsedesy** [comment]
32.90 is another extraneous empty line. The lines 85–90 are out of ordered and manually numbered, but it a `<lb rend="displayNum" n="90" />` was inserted somehow as well, effectively creating a blank line numbered 90 in the middle. Fixed in f9622675f216697a842aa92e89cba0f2715d5a8c.

```
<lb rend="displayNum" n="84" />gai=a de\ khw/essan a)naptu/casa loxei/hn
<lb rend="displayNum" n="85" />a)/nqesin i(mertoi=si gamh/lion e)/stefen eu)nh/n:
<lb rend="displayNum" n="86" />kai\ kro/kos e)bla/sthse *ki/lic kai\ e)fu/eto mi=lac,
<lb rend="displayNum" n="88" />qh/lei+ d' a)/rsena fu/lla sune/pleke gei/toni poi/h|,
<lb rend="displayNum" n="89" /><lb rend="displayNum" n="90" />oi(=a po/qou pnei/wn kai\ e)n a)/nqesin a(bro\s a)koi/ths,
<lb rend="displayNum" n="87" />kai\ le/xos a)mfote/rwn e)peko/smee diplo/os o)/rphc,
<lb rend="displayNum" n="90" />*zh=na kro/kw| puka/sas kai\ mi/laki su/ggamon *(/hrhn:
<lb rend="displayNum" n="91" />kai\ *dio\s o)cu\n e)/rwta noh/moni dei/knue sigh=|
```

https://archive.org/details/dionysiaca02nonnuoft/page/450/mode/1up
![dionysiaca02nonnuoft_0466](https://user-images.githubusercontent.com/53617134/210245173-631966d9-c196-4cf1-8d2d-b270bc98b54c.jpg)

Perseus 5.0 has the same error:
https://github.com/PerseusDL/canonical-greekLit/blob/812f91f083f88cf789b37be89b22606ca6f27f6c/data/tlg2045/tlg001/tlg2045.tlg001.perseus-grc1.xml
```
<l n="84">γαῖα δὲ κηώεσσαν ἀναπτύξασα λοχείην</l>
<l n="85">ἄνθεσιν ἱμερτοῖσι γαμήλιον ἔστεφεν εὐνήν· </l>
<l n="86">καὶ κρόκος ἐβλάστησε Κίλιξ καὶ ἐφύετο μῖλαξ, </l>
<l n="88">θήλεϊ δʼ ἄρσενα φύλλα συνέπλεκε γείτονι ποίῃ, </l>
<l n="89"/>
<l n="90">οἷα πόθου πνείων καὶ ἐν ἄνθεσιν ἁβρὸς ἀκοίτης, </l>
<l n="87">καὶ λέχος ἀμφοτέρων ἐπεκόσμεε διπλόος ὄρπηξ, </l>
<l n="90">Ζῆνα κρόκῳ πυκάσας καὶ μίλακι σύγγαμον Ἥρην· </l>
<l n="91">καὶ Διὸς ὀξὺν ἔρωτα νοήμονι δείκνυε σιγῇ</l>
```

**Comment by whoopsedesy** [comment]
The duplicate 40.566 looks like it's just an error, perhaps caused by confusion around out-of-order line numbers and a typographically split line. The second instance of 40.566 in the TEI should have been 40.567 to match the printed version. Fixed in 7574d199148492ba94a64a31dd40f5d3becd926a.

https://archive.org/details/dionysiaca03nonnuoft/page/192/mode/1up
![dionysiaca03nonnuoft_0208](https://user-images.githubusercontent.com/53617134/210251514-9584f97b-87a9-42f7-8a98-42fd39621388.jpg)

Perseus 5.0 has the same error:
https://github.com/PerseusDL/canonical-greekLit/blob/812f91f083f88cf789b37be89b22606ca6f27f6c/data/tlg2045/tlg001/tlg2045.tlg001.perseus-grc1.xml

```
<l n="564">Ἄρτεμις οὐ βλάστησεν ἀφʼ ὕδατος, ὡς Ἀφροδίτη. </l>
<l n="566">ἔννεπε Καλλιρόῃ· Δροσερῇ μὴ κρύπτε καὶ αὐτῇ. </l>
<l n="565">Κύπριδι μᾶλλον ὄφελλες ἄγειν χάριν, ὅττι καὶ αὐτὴ </l>
<l n="566">αὐχένα κάμψεν Ἔρωτι, καὶ εἰ τροφός ἐστιν Ἐρώτων.</l>
<l n="567">δέχνυσο κέντρα πόθοιο, καὶ ὑγρονόμον σε καλέσσω</l>
```

Aside, I'll note that the typographic split of 40.567 is not represented in the TEI, as the one at 37.625 is.

**Comment by whoopsedesy** [comment]
The cause of duplicate 37.625 is clear: it's a typographically split line at 37.621 whose two parts were wrongly given separate line numbers 621 and 622. What's called "622" should be the remainder of 621, and the following lines up to 625 shifted down a place.

There are two options for resolving it. @sasansom please advise.

1. Keep the line break and number both partial lines 621. (As in #27.)
   ```
   <lb rend="displayNum" n="621" />w(\s fame/nou
   <lb rend="displayNum" n="622" />*diktai=os e)qh/mona gou/nata pa/llwn...
   ```
2. Remove the line break and make it one line in the TEI. (As with [40.567](https://github.com/sasansom/sedes/issues/79#issuecomment-1369039631).)
   ```
   <lb rend="displayNum" n="621" />w(\s fame/nou *diktai=os e)qh/mona gou/nata pa/llwn...
   ```

In either case, we will also need a change in src/known.py. Currently [it has the first part of 37.621 but not the second](https://github.com/sasansom/sedes/blob/7574d199148492ba94a64a31dd40f5d3becd926a/src/known.py#L3081-L3082):

```py
    'ὣς φαμένου': # Dion. 37.621
        (('ὣς', '+'), ('φαμένου', '--+')),
```

This is causing the second part to be scanned as a four-word all-spondee line:

```
Dion.,37,621,1,ὣς,ὡς,1,–,manual,1,ὣς φαμένου
Dion.,37,621,2,φαμένου,φημί,2,⏑⏑–,manual,1,ὣς φαμένου
Dion.,37,622,1,δικταῖος,δικταῖος,1,–––,auto,1,Δικταῖος ἐθήμονα γούνατα πάλλων...
Dion.,37,622,2,ἐθήμονα,ἐθήμων,4,––––,auto,1,Δικταῖος ἐθήμονα γούνατα πάλλων...
Dion.,37,622,3,γούνατα,γόνυ,8,–––,auto,1,Δικταῖος ἐθήμονα γούνατα πάλλων...
Dion.,37,622,4,πάλλων,πάλλω,11,––,auto,1,Δικταῖος ἐθήμονα γούνατα πάλλων...
```

With option (1), we'll need the second part of 37.621 to be added to src/known.py. With option (2), perhaps automatic scansion, having the whole line to work with, will get it right and we can remove the entry from src/known.py.

https://archive.org/details/dionysiaca03nonnuoft/page/78/mode/1up
![dionysiaca03nonnuoft_0094](https://user-images.githubusercontent.com/53617134/210255383-457d1550-c8af-421f-b0fd-f1856f019542.jpg)

https://archive.org/details/dionysiaca03nonnuoft/page/80/mode/1up
![dionysiaca03nonnuoft_0096](https://user-images.githubusercontent.com/53617134/210255401-db769857-e23a-424e-b59e-ef0799e5bd4c.jpg)

Perseus 5.0 has the same error:

https://github.com/PerseusDL/canonical-greekLit/blob/812f91f083f88cf789b37be89b22606ca6f27f6c/data/tlg2045/tlg001/tlg2045.tlg001.perseus-grc1.xml
```
<l n="621">ὣς φαμένου</l>
<l n="622">Δικταῖος ἐθήμονα γούνατα πάλλων...</l>
<pb id="v.3.p.80"/>
<l n="623">τῷ δʼ ἐπὶ ποικιλόμητις ἀνέδραμεν ὠκὺς Ἐρεχθεύς,</l>
<l n="624">Παλλάδι Νικαίῃ μεμελημένος, αὐτὰρ ἐπʼ αὐτῷ</l>
<l n="625">Πρίασος ὠκυπόδης, Κυβεληίδος ἀστὸς ἀρούρης.</l>
<l n="625">τοῖσι μὲν ἐκ βαλβῖδος ἔην δρόμος· Ὠκύθοος δὲ </l>
```

**Comment by whoopsedesy** [comment]
39.385 is fixed in commit 57971eb20b4c8b579227a75d1bb42bbdc8d578d5, but it deserves more comment.

The cause appears to be the same as with [37.625](https://github.com/sasansom/sedes/issues/79#issuecomment-1369059625), an earlier line that is typographically split. The two consecutive lines that have printed markers 380 and 382:

> καὶ Ζέφυρος κεκόρυστο,
> Νότος δʼ ἐπεσύρισεν Εὔρῳ,

are supposed to be only one metrical line.

The two lines were united in commit https://github.com/sasansom/sedes/commit/2b4737c73eadf5c0c41233ef3e63c21f7463197a#diff-c4e168f0263854513ee388716b85afe7d1dc35e0efe75cf605a6f3d4e179d920L17325-R17324; however, the numbering of the following lines were not appropriately adjusted (shifting 383 down to 382, 384 down to 383, 385 down to 384), which is why there was still a duplicate line warning. I guess we are going with the explanation that the printed line number 382 is a misprint that should have been placed one line lower. I made the line number adjustment in 57971eb20b4c8b579227a75d1bb42bbdc8d578d5.

I will note that the printed version has a capital Νότος, while we and Perseus 5.0 have a lower-case νότος. @sasansom, is this an error?

https://archive.org/details/dionysiaca03nonnuoft/page/150/mode/1up
![dionysiaca03nonnuoft_0166](https://user-images.githubusercontent.com/53617134/210258117-d7f1d86f-f63c-41d7-ba4a-6edab0245ca2.jpg)

The error is still present in Perseus 5.0. Already noted at #53 ("`2b4737c` nonnusdionysiaca.xml@19397").

https://github.com/PerseusDL/canonical-greekLit/blob/812f91f083f88cf789b37be89b22606ca6f27f6c/data/tlg2045/tlg001/tlg2045.tlg001.perseus-grc1.xml
```
<l n="379">δυσμενέων ἐθέλοντες ἀιστῶσαι στίχα νηῶν, </l>
<l n="381">οἱ μὲν Δηριαδῆος ἀρηγόνες, οἱ δὲ Λυαίου· </l>
<l n="380">καὶ Ζέφυρος κεκόρυστο, </l>
<l n="382">νότος δʼ ἐπεσύρισεν Εὔρῳ, </l>
<l n="383">καὶ Βορέης Θρήισσαν ἄγων ἀντίπνοον αὔρην</l>
<l n="384">ἄγρια μαινομένης ἐπεμάστιε νῶτα θαλάσσης.</l>
<l n="385">καὶ στόλον ἰθύνουσα μαχήμονα Δηριαδῆος</l>
<l n="385">ὑσμίνης Ἔρις ἦρχε· Διωνύσοιο δὲ νηῶν </l>
```

**Comment by whoopsedesy** [comment]
I have fixed the ones that were clear and simple errors. Besides [37.625](https://github.com/sasansom/sedes/issues/79#issuecomment-1369059625), which has two possible solutions, @sasansom I need your attention on 37.568 and 44.145.

## 37.568

I don't see a way to fix 37.568. On page 74, there is a line that follows a printed line number 567 (and is therefore implicitly numbered 568); then on page 78 there is a printed line number 568.

The implicitly numbered 568 precedes a line with printed line number 572; but cannot be that the line is rather supposed to be 571, because there is also another line 571 later on, on page 78.

|effective line #|printed line #|*|
|--:|--:|---|
|565|565|near bottom of page 74|
|566|||
|567|567||
|568||*|
|572|572|top of page 76|
|573|||
|575|575||
|…|…||
|600|600|top of page 78|
|601|601||
|568|568|*|
|569|569||
|570|570||
|571|571||
|602|602||
|603|||
|604|||
|605|605||

https://archive.org/details/dionysiaca03nonnuoft/page/74/mode/1up
![dionysiaca03nonnuoft_0090](https://user-images.githubusercontent.com/53617134/210259941-2daa78f0-fac4-4dce-bca3-7fcd4b161cd6.jpg)

https://archive.org/details/dionysiaca03nonnuoft/page/76/mode/1up
![dionysiaca03nonnuoft_0092](https://user-images.githubusercontent.com/53617134/210259966-156510f5-22b7-4062-9cc3-cff533f204ca.jpg)

https://archive.org/details/dionysiaca03nonnuoft/page/78/mode/1up
![dionysiaca03nonnuoft_0094](https://user-images.githubusercontent.com/53617134/210259981-7144b19f-3823-4788-9d46-05a3f0c3193b.jpg)

Perseus 5.0 has the same problem. (It also happens to be missing line 37.600, which I guess is unrelated.)

https://github.com/PerseusDL/canonical-greekLit/blob/812f91f083f88cf789b37be89b22606ca6f27f6c/data/tlg2045/tlg001/tlg2045.tlg001.perseus-grc1.xml
```
<l n="565">μεσσατίῳ δὲ κάρηνον ἐπηρείδοντο μετώπῳ </l>
<l n="566">ἀκλινέες, νεύοντες ἐπὶ χθονός· ἐκ δὲ μετώπων</l>
<l n="567">θλιβομένων καμάτοιο προάγγελος ἔρρεεν ἱδρώς· </l>
<l n="568">ἀμφοτέρων δʼ ἄρα νῶτα κεκυφότα πήχεος ὁλκῷ</l>
<pb id="v.3.p.76"/>
<l n="572">δίζυγι συμπλεκέος παλάμης ἐτρίβετο δεσμῷ· </l>
<l n="573">σμῶδιξ δʼ αὐτοτέλεστος ἀνέδραμεν αἵματι θερμῷ,</l>
...
<l n="599">καὶ ταχὺς ἀντιβίου τετανυσμένος ὑψόθι νώτων,</l>
<pb id="v.3.p.78"/>
<l n="601">αὐχένι δεσμὸν ἔβαλλε βραχίονι, δάκτυλα κάμψας· </l>
<l n="568">μυδαλέῳ δʼ ἱδρῶτι χυτὴν ἔρραινε κονίην, </l>
<l n="569">αὐχμηρῇ ψαμάθῳ διερὴν ῥαθάμιγγα καθαίρων, </l>
<l n="570">μὴ διολισθήσειε περίπλοκος ἅμματι χειρῶν </l>
<l n="571">θερμὴν τριβομένοιο κατʼ αὐχένος ἰκμάδα πέμπων. </l>
<l n="602">τοῦ δὲ πιεζομένοιο συνέρρεον ὀξέι παλμῷ </l>
```

## 44.145

There are 5 unnumbered lines between 140 and 145 in the book scan; then there is no line 147. Perhaps another misprint?

https://archive.org/details/dionysiaca03nonnuoft/page/306/mode/1up
![dionysiaca03nonnuoft_0322](https://user-images.githubusercontent.com/53617134/210260069-faa52808-21b3-481c-b4ad-4f7d5b9c34fe.jpg)

https://archive.org/details/dionysiaca03nonnuoft/page/308/mode/1up
![dionysiaca03nonnuoft_0324](https://user-images.githubusercontent.com/53617134/210260078-2d290bea-c99f-4373-ad47-bde2eadc6ca2.jpg)

Perseus 5.0 has the same problem:

https://github.com/PerseusDL/canonical-greekLit/blob/812f91f083f88cf789b37be89b22606ca6f27f6c/data/tlg2045/tlg001/tlg2045.tlg001.perseus-grc1.xml
```
<l n="140">κύμβαλα δʼ ἠχήεντα διαρρίψαντες ἀήταις </l>
<l n="141">καὶ πάταγον Βερέκυντα καὶ Εὔια τύμπανα Ῥείης</l>
<l n="142">ἕλκετε Βασσαρίδας μανιώδεας, ἕλκετε Βάκχας,</l>
<l n="143">ἀμφιπόλους Βρομίοιο συνήλυδας, ἃς ἐνὶ Θήβῃ</l>
<pb id="v.3.p.308"/>
<l n="144">Ἰσμηνοῦ διεροῖσιν ἀκοντίζοντες ἐναύλοις</l>
<l n="145">Νηίδας Ἀονίαις ποταμηίσι μίξατε Νύμφαις</l>
<l n="145">ἥλικας, Ἁδρυάδας δὲ γέρων δέξαιτο Κιθαιρὼν </l>
<l n="146">ἄλλαις Ἁδρυάδεσσιν ὁμόζυγας ἀντὶ Δυαίου. </l>
<l n="148">ἄξατε πῦρ, θεράποντες, ἐπεὶ ποινήτορι θεσμῷ, </l>
<l n="149">ἐκ πυρὸς εἰ πέλε Βάκχος, ἐγὼ πυρὶ Βάκχον ὀπάσσω·</l>
<l n="150">Ζεὺς Σεμέλην ἐδάμασσεν, ἐγὼ Διόνυσον ὀλέσσω. </l>
```

**Comment by sasansom** [comment]
For Dion. 37.621 (discussed [here](https://github.com/sasansom/sedes/issues/79#issuecomment-1369059625)), we should go with option 2. I have fixed this now in commit [bc0e28f](https://github.com/sasansom/sedes/commit/bc0e28f20c6e603c2721075b957bb5a81a6fc62c). Hopefully the automatic scansion will clear things up. I've deleted its entry in known.py (commit 1466dff).

**Comment by sasansom** [comment]
> 39.385 is fixed in commit [57971eb](https://github.com/sasansom/sedes/commit/57971eb20b4c8b579227a75d1bb42bbdc8d578d5), but it deserves more comment.
> 
> The cause appears to be the same as with [37.625](https://github.com/sasansom/sedes/issues/79#issuecomment-1369059625), an earlier line that is typographically split. The two consecutive lines that have printed markers 380 and 382:
> 
> > καὶ Ζέφυρος κεκόρυστο,
> > Νότος δʼ ἐπεσύρισεν Εὔρῳ,
> 
> are supposed to be only one metrical line.
> 
> The two lines were united in commit [2b4737c#diff-c4e168f0263854513ee388716b85afe7d1dc35e0efe75cf605a6f3d4e179d920L17325-R17324](https://github.com/sasansom/sedes/commit/2b4737c73eadf5c0c41233ef3e63c21f7463197a#diff-c4e168f0263854513ee388716b85afe7d1dc35e0efe75cf605a6f3d4e179d920L17325-R17324); however, the numbering of the following lines were not appropriately adjusted (shifting 383 down to 382, 384 down to 383, 385 down to 384), which is why there was still a duplicate line warning. I guess we are going with the explanation that the printed line number 382 is a misprint that should have been placed one line lower. I made the line number adjustment in [57971eb](https://github.com/sasansom/sedes/commit/57971eb20b4c8b579227a75d1bb42bbdc8d578d5).
> 
> I will note that the printed version has a capital Νότος, while we and Perseus 5.0 have a lower-case νότος. @sasansom, is this an error?

Yes, I think it's an error. I've now capitalized Notos in commit [55fcfb4](https://github.com/sasansom/sedes/commit/55fcfb47f40e737482e222462dd1cd168c4ad79e).

**Comment by sasansom** [comment]
> I have fixed the ones that were clear and simple errors. Besides [37.625](https://github.com/sasansom/sedes/issues/79#issuecomment-1369059625), which has two possible solutions, @sasansom I need your attention on 37.568 and 44.145.
> 
> ## 37.568
> I don't see a way to fix 37.568. On page 74, there is a line that follows a printed line number 567 (and is therefore implicitly numbered 568); then on page 78 there is a printed line number 568.
> 
> The implicitly numbered 568 precedes a line with printed line number 572; but cannot be that the line is rather supposed to be 571, because there is also another line 571 later on, on page 78.
> 
What's going on here is 1) Rouse, the editor of the Loeb edition, chose to transpose lines 37.568–71 to after 37.601 and before 37.602 (p. 78), but 2) at the top of p. 76 the edition incorrectly printed 572. It should have printed 573, or better yet, have printed 572 on the bottom of p. 74 (which I assume was the original intention before typesetting; it also corresponds to 37.572 in Keydell's edition [1959]).

And there is separate problem. The xml file does not include 37.600 anywhere in the text (corresponding to 600 on top of p.78); it seems to have dropped out entirely. I have now added it.

I have fixed both in commit b180f53.

> ## 44.145
> There are 5 unnumbered lines between 140 and 145 in the book scan; then there is no line 147. Perhaps another misprint?

Here Rouse follows the edition of Ludwich (Teubner 1911) which transposes line 147 between 138 and 139.

<img width="474" alt="Screen Shot 2023-01-02 at 9 19 46 PM" src="https://user-images.githubusercontent.com/6109788/210292495-8c57bfd3-87ba-47f6-84fe-ac3d447c3856.png">

The problem is that the Loeb edition prints the verse numbers incorrectly; the sequence [138, 147, 139, 140] on p. 306 should be shifted down one line.

I've fixed this in commit 7c1e467.

**Comment by whoopsedesy** [comment]
I've made sure that the commits resulting from this issue are reflected at #53.
